src/s1reader/s1_annotation.py

```diff
@@ -467,7 +468,58 @@ def from_noise_annotation(cls, noise_annotation: NoiseAnnotation, azimuth_time:
                    azimuth_first_azimuth_line, azimuth_first_range_sample,
                    azimuth_last_azimuth_line, azimuth_last_range_sample,
                    azimuth_line, azimuth_lut,
-                   line_from, line_to)
+                   line_from, line_to, ipf_version)
+
+    def compute_burst_thermal_noise_lut(self, burst_shape):
+        '''Returns the burst-sized LUT for thermal noise correction
+
+        Parameters
+        ----------
+        burst_shape: tuple[int, int]
+            Shape of burst and corresponding thermal noise LUT
+
+        Returns
+        -------
+        arr_lut_total: np.array
+            2d array containing thermal noise correction look up table values
+        '''
+        # check IPF version
+        if not self.can_compute_thermal_noise_lut():
+            print("IPF version doesn't no support thermal noise")
+            return np.ones(burst_shape)
+
+        nrows, ncols = burst_shape
+
+        # Interpolate the azimuth noise vector
+        az_lut_interp_obj = InterpolatedUnivariateSpline(self.azimuth_line,
+                                                         self.azimuth_lut,
+                                                         k=1)
+        vec_az = np.arange(self.line_from, self.line_to + 1)
+        az_lut_interpolated = az_lut_interp_obj(vec_az)
+
+        # Interpolate the range noise vector
+        rg_lut_interp_obj = InterpolatedUnivariateSpline(self.range_pixel,
+                                                         self.range_lut,
+                                                         k=1)
+        vec_rg = np.arange(self.azimuth_last_range_sample + 1)
+        rg_lut_interpolated = rg_lut_interp_obj(vec_rg)
+
+        arr_lut_total = np.matmul(az_lut_interpolated[..., np.newaxis],
+                                  rg_lut_interpolated[np.newaxis, ...])
+        return arr_lut_total
+
+    def can_compute_thermal_noise_lut(self):
+        '''
+        Determine if class instance IPF version supports azimuth noise vector
+
+        Returns
+        -------
+        _: bool
+            True if thermal noise vector is supported, otherwise False.
+        '''
+        if self.ipf_version < ipf_version_az_noise_vector_available_from:
+            return False
+        return True
```

Comment:
So it seems like `BurstNoise.can_compute_thermal_noise_lut()` returns `False` when `ipf_version` of the burst is below the threshold (i.e. `ipf_version_az_noise_vector_available_from`), so that `compute_burst_thermal_noise(()` returns the array of ones.

That is different to what I initially implemented. There are two components of noise vector: `range noise vector`, and `azimuth noise vector`. NADS with IPF version `<2.90` does not provide azimuth noise vector, whereas the ones with `>=2.90` do provide that. My intention was to apply the azimuth noise vector when it's available in NADS, or just set the vector to ones when we don't have it.